producer-kafka/src/test/java/io/stargate/producer/kafka/KafkaCDCProducerIntegrationTest.java

```diff
@@ -16,76 +16,78 @@
-import static org.assertj.core.api.Assertions.assertThatCode;
 import static org.awaitility.Awaitility.await;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
+import static org.testcontainers.containers.KafkaContainer.ZOOKEEPER_PORT;
 
 import com.google.common.collect.Streams;
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+import io.confluent.kafka.serializers.KafkaAvroSerializer;
 import io.stargate.producer.kafka.configuration.ConfigLoader;
-import io.stargate.producer.kafka.schema.KeyValueConstructor;
-import io.stargate.producer.kafka.schema.MockKafkaAvroSerializer;
-import io.stargate.producer.kafka.schema.MockKeyKafkaAvroDeserializer;
-import io.stargate.producer.kafka.schema.MockValueKafkaAvroDeserializer;
-import io.stargate.producer.kafka.schema.SchemaProvider;
+import io.stargate.producer.kafka.schema.EmbeddedSchemaRegistryServer;
+import java.net.ServerSocket;
 import java.time.Duration;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Properties;
 import java.util.UUID;
-import org.apache.avro.generic.GenericData;
 import org.apache.avro.generic.GenericRecord;
 import org.apache.cassandra.stargate.db.DeleteEvent;
 import org.apache.cassandra.stargate.db.RowUpdateEvent;
+import org.apache.cassandra.stargate.schema.CQLType.Native;
 import org.apache.cassandra.stargate.schema.TableMetadata;
 import org.apache.kafka.clients.consumer.ConsumerConfig;
 import org.apache.kafka.clients.consumer.ConsumerRecords;
 import org.apache.kafka.clients.consumer.KafkaConsumer;
 import org.apache.kafka.clients.producer.ProducerConfig;
-import org.apache.kafka.common.errors.TimeoutException;
 import org.jetbrains.annotations.NotNull;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.testcontainers.containers.KafkaContainer;
 import org.testcontainers.containers.Network;
-import org.testcontainers.containers.ToxiproxyContainer;
-import org.testcontainers.containers.ToxiproxyContainer.ContainerProxy;
 
 class KafkaCDCProducerIntegrationTest {
 
-  private static int PORT = 9093;
   private static KafkaContainer kafkaContainer;
-  private static ToxiproxyContainer toxiproxyContainer;
-  private static ContainerProxy kafkaProxy;
+  private static EmbeddedSchemaRegistryServer schemaRegistry;
 
   private static final String TOPIC_PREFIX = "topicPrefix";
 
   @BeforeAll
-  public static void setup() {
+  public static void setup() throws Exception {
     Network network = Network.newNetwork();
-    kafkaContainer = new KafkaContainer().withNetwork(network);
-    toxiproxyContainer = new ToxiproxyContainer().withNetwork(network);
-    toxiproxyContainer.start();
-    kafkaProxy = toxiproxyContainer.getProxy(kafkaContainer, PORT);
+    kafkaContainer = new KafkaContainer().withNetwork(network).withEmbeddedZookeeper();
```

Comment:
oh, the testcontainers API for kafka is super comfortable.

Reply:
yes, it has great Kafka integration (only schema-registry is missing, but it is very easy to setup and start embedded version)